dipy/testing/decorators.py

```diff
@@ -103,3 +109,109 @@ def _set_random_number_generator_wrapper(pytestconfig, *args, **kwargs):
         return _set_random_number_generator_wrapper
 
     return _set_random_number_generator
+
+
+def warning_for_keywords(from_version="1.10.0", until_version="2.0.0"):
+    """
+    Decorator to warn about keyword arguments passed as positional arguments
+    and handle version-based deprecation of functions.
+
+    Parameters
+    ----------
+    from_version : str
```

Comment:
from_version will be: Optional, str intead of str

Reply:
indeed,   `str, optional` instead